app/src/main/java/org/kiwix/kiwixmobile/main/MainActivity.java

```diff
@@ -330,81 +332,44 @@ public void onCreate(Bundle savedInstanceState) {
 
-    initPlayStoreUri();
-    isHideToolbar = sharedPreferenceUtil.getPrefHideToolbar();
-
-    FileReader fileReader = new FileReader();
-    documentParserJs = fileReader.readFile("js/documentParser.js", this);
-    documentSections = new ArrayList<>();
-    tabsAdapter = new TabsAdapter(this, webViewList);
-    tabsAdapter.registerAdapterDataObserver(new RecyclerView.AdapterDataObserver() {
-      @Override
-      public void onChanged() {
-        updateTabSwitcherIcon();
-      }
-    });
+    RecyclerView tableDrawerRight = tableDrawerRightContainer.getHeaderView(0).findViewById(R.id.right_drawer_list);
 
-    tableDrawerRight.setLayoutManager(new LinearLayoutManager(this));
+    tableDrawerRight = tableDrawerRightContainer.getHeaderView(0).findViewById(R.id.right_drawer_list);
 
-    TableDrawerAdapter tableDrawerAdapter = setupTableDrawerAdapter();
-    tableDrawerRight.setAdapter(tableDrawerAdapter);
-    tableDrawerAdapter.notifyDataSetChanged();
+    checkForRateDialog();
 
-    tabsAdapter.setTabClickListener(new TabsAdapter.TabClickListener() {
-      @Override
-      public void onSelectTab(View view, int position) {
-        hideTabSwitcher();
-        selectTab(position);
+    initPlayStoreUri();
+    isHideToolbar = sharedPreferenceUtil.getPrefHideToolbar();
 
-        /* Bug Fix
-         * Issue #592 in which the navigational history of the previously open tab (WebView) was
-         * carried forward to the newly selected/opened tab; causing erroneous enabling of
-         * navigational buttons.
-         */
-        updateBottomToolbarArrowsAlpha();
-      }
+    addFileReader();
+    setupTabsAdapter();
+    setTableDrawerInfo();
+    setTabListener();
 
-      @Override
-      public void onCloseTab(View view, int position) {
-        closeTab(position);
-      }
-    });
     compatCallback = new CompatFindActionModeCallback(this);
     setUpTTS();
-    documentParser = new DocumentParser(new DocumentParser.SectionsListener() {
-      @Override
-      public void sectionsLoaded(String title, List<DocumentSection> sections) {
-        for (DocumentSection section : sections) {
-          if (section.title.contains("REPLACE_")) {
-            section.title = getResourceString(getApplicationContext(), section.title);
-          }
-        }
-        documentSections.addAll(sections);
-        if (title.contains("REPLACE_")) {
-          tableDrawerAdapter.setTitle(getResourceString(getApplicationContext(), title));
-        } else {
-          tableDrawerAdapter.setTitle(title);
-        }
-        tableDrawerAdapter.setSections(documentSections);
-        tableDrawerAdapter.notifyDataSetChanged();
-      }
 
-      @Override
-      public void clearSections() {
-        documentSections.clear();
-        tableDrawerAdapter.notifyDataSetChanged();
-      }
-    });
+    setupDocumentParser();
 
     manageExternalLaunchAndRestoringViewState();
     loadPrefs();
     updateTitle();
 
-    Intent i = getIntent();
+
```

Comment:
You should remove this blank line as there are two blank lines. The best way to avoid these type of mistakes is to import Square's code style as described in CODE_STYLE.md and reformat the code before committing using `Ctrl + Shift + L` or from `Code -> Reformat Code` but the easiest way is to go to menu select `VCS -> Git -> Commit File` and select the files to be committed and make the top 3 options on the right side checked (make sure to import Square's code style before doing this otherwise Android's default code style will be used):
![image](https://user-images.githubusercontent.com/20206262/52482996-c33d1000-2bd8-11e9-9b64-38072eb60751.png)

Reply:
@abdulwd I've made all these changes now and also I've tried to squash the commits. Please check if I've squashed accurately.